src/workerd/api/streams/internal.c++

```diff
@@ -515,31 +515,35 @@ kj::Maybe<jsg::Promise<ReadResult>> ReadableStreamInternalController::read(
         js.v8TypeError("This ReadableStream belongs to an object that is closing."_kj));
   }
 
-  std::shared_ptr<v8::BackingStore> store;
+  v8::Local<v8::ArrayBuffer> store;
   size_t byteLength = 0;
   size_t byteOffset = 0;
   size_t atLeast = 1;
 
   KJ_IF_SOME(byobOptions, maybeByobOptions) {
-    auto buffer = byobOptions.bufferView.getHandle(js)->Buffer();
-    store = buffer->GetBackingStore();
+    store = byobOptions.bufferView.getHandle(js)->Buffer();
     byteOffset = byobOptions.byteOffset;
     byteLength = byobOptions.byteLength;
     atLeast = byobOptions.atLeast.orDefault(atLeast);
     if (byobOptions.detachBuffer) {
-      if (!buffer->IsDetachable()) {
+      if (!store->IsDetachable()) {
         return js.rejectedPromise<ReadResult>(
             js.v8TypeError("Unable to use non-detachable ArrayBuffer"_kj));
       }
-      jsg::check(buffer->Detach(v8::Local<v8::Value>()));
+      auto backing = store->GetBackingStore();
+      jsg::check(store->Detach(v8::Local<v8::Value>()));
+      store = v8::ArrayBuffer::New(js.v8Isolate, kj::mv(backing));
     }
   }
 
   auto getOrInitStore = [&](bool errorCase = false) {
-    if (!store) {
+    if (store.IsEmpty()) {
       // In an error case, where store is not provided, we can use zero length
       byteLength = errorCase ? 0 : UnderlyingSource::DEFAULT_AUTO_ALLOCATE_CHUNK_SIZE;
-      store = v8::ArrayBuffer::NewBackingStore(js.v8Isolate, byteLength);
+
+      if (!v8::ArrayBuffer::MaybeNew(js.v8Isolate, byteLength).ToLocal(&store)) {
```

Comment:
Why do we use `MaybeNew` here but `New` [here](https://github.com/cloudflare/workerd/pull/2185/files#diff-745dea5c7b6cdcab0023d80592ab9318c5a8adbf6e40e83e73e66b7d86241f74R535)?

Reply:
If I'm following the link there correctly, this one is actually allocating new space, the other is just wrapping around the existing allocation.

Reply:
Ah, of course, so the one that wraps an existing allocation can't fail which is why it's `New` and not `MaybeNew`. Makes sense.